protocols/messages_test.go

```diff
@@ -103,3 +105,47 @@ func TestMessage(t *testing.T) {
 	})
 
 }
+
+func TestSortedProposals(t *testing.T) {
+	type TestCase struct {
+		Input       []consensus_channel.SignedProposal
+		Expectation []consensus_channel.SignedProposal
+	}
+	testCases := []TestCase{
+		{
+			Input: []consensus_channel.SignedProposal{
+				addProposal(types.Destination{'a'}, 1),
+				addProposal(types.Destination{'a'}, 0),
```

Comment:
Obviously these are well defined in context of the unit under test (the sort algo), but it's maybe noteworthy that a ledger channel should never fund the same channelID more than once. I don't know whether it'd be worthwhile for a data structure at some point to prevent it.

Reply:
It isn't very clear from reading the code, but these destinations do not represent target (virtual) channel Ids. They represent the id of the ledger channel. So they aren't falling foul of the rule you mention.

Reply:
Correction: perhaps that isn't what you meant. The `addProposal` helper uses a fixed guarantee targeting a fixed channel, so the proposals _are_ falling foul of that rule, independent of the arguments we pass in to `addProposal`.